creators/Fenix/StopsCreatorFenix.py

```diff
@@ -1,19 +1,150 @@
 # coding=utf-8
 
+import overpy
+import sys
+
+from transitfeed import util
 from creators.StopsCreator import StopsCreator
+from osmhelper.osm_routes import Route, RouteMaster
+from osmhelper.osm_stops import Stop
+from osmhelper.cache import Cache
 
 
 class StopsCreatorFenix(StopsCreator):
 
     def add_stops_to_schedule(self, schedule, data):
 
-        stops = data.stops
+        # Get stops information
+        stops = data.get_stops()
+        stops_changed = False
 
         # add all stops to GTFS
         for stop in stops.values():
+
+            # If there is no name, query one intelligently from OSM
+            if stop.name == Stop.NO_NAME:
+                self.find_best_name_for_unnamed_stop(stop)
+                stops_changed = True
+                print stop
+
+            # Add stop to GTFS object
             schedule.AddStop(
                 lat=float(stop.lat),
                 lng=float(stop.lon),
                 name=stop.name,
                 stop_id=str(stop.id)
             )
+
+        if stops_changed:
+            # Cache stops with newly created stop names
+            Cache.write_data('stops', self.stops)
```

Comment:
Yes, good idea to cache every time a name has been found (successfully or not), to be able to continue. This also would allow to get rid of the `stops_changed` flag. I love it. Changed in latest update to the commit.